pkg/apis/controller/common/v1beta1/common_types.go

```diff
@@ -27,6 +27,8 @@ type AlgorithmSpec struct {
 	AlgorithmName string `json:"algorithmName,omitempty"`
 
 	// Key-value pairs representing settings for suggestion algorithms.
+	// +listType=map
+	// +listMapKey=Name
```

Comment:
Should this Name be lowercase, like here: https://github.com/kubeflow/training-operator/blob/master/pkg/apis/kubeflow.org/v2alpha1/trainjob_types.go#L210 ?